WordPressKit/WordPressOrgXMLRPCApi.swift

```diff
@@ -332,13 +387,13 @@ open class WordPressOrgXMLRPCApi: NSObject {
     @objc public static let WordPressOrgXMLRPCApiErrorKeyDataString: NSError.UserInfoKey = "WordPressOrgXMLRPCApiErrorKeyDataString"
     @objc public static let WordPressOrgXMLRPCApiErrorKeyStatusCode: NSError.UserInfoKey = "WordPressOrgXMLRPCApiErrorKeyStatusCode"
 
-    private func convertError(_ error: NSError, data: Data?, statusCode: Int? = nil) -> NSError {
+    fileprivate static func convertError(_ error: NSError, data: Data?, statusCode: Int? = nil) -> NSError {
         let responseCode = statusCode == 403 ? 403 : error.code
         if let data = data {
-            var userInfo: [AnyHashable: Any] = error.userInfo
-            userInfo[type(of: self).WordPressOrgXMLRPCApiErrorKeyData] = data
-            userInfo[type(of: self).WordPressOrgXMLRPCApiErrorKeyDataString] = NSString(data: data, encoding: String.Encoding.utf8.rawValue)
-            userInfo[type(of: self).WordPressOrgXMLRPCApiErrorKeyStatusCode] = statusCode
+            var userInfo: [String: Any] = error.userInfo
+            userInfo[Self.WordPressOrgXMLRPCApiErrorKeyData as String] = data
+            userInfo[Self.WordPressOrgXMLRPCApiErrorKeyDataString as String] = NSString(data: data, encoding: String.Encoding.utf8.rawValue)
+            userInfo[Self.WordPressOrgXMLRPCApiErrorKeyStatusCode as String] = statusCode
```

Comment:
I was surprised by the `as String` so I drilled into the type definition. All those keys are `NSError.UserInfoKey` which is an `NSString`.

I wonder why Apple decided to define `userInfo` keys as `NSString` when other similar entities are `String`s:

![image](https://github.com/wordpress-mobile/WordPressKit-iOS/assets/1218433/afeb3ceb-e58c-47e4-8bd5-bdb441913574)

Reply:
Side note: From the quick look I had at the code, I got the impression all usages ended up cast as `String`.

Unless I'm wrong, we could followup with a simplification PR that defines those keys as `String`.